modepy/quadrature/__init__.py

```diff
@@ -112,3 +132,54 @@ def __init__(self, N, dims, backend=None):      # noqa: N803
         from modepy.quadrature.jacobi_gauss import LegendreGaussQuadrature
         super().__init__(
                 dims, LegendreGaussQuadrature(N, backend=backend))
+
+
+# {{{ quadrature
+
+@singledispatch
+def quadrature_for_space(space: FunctionSpace, shape: Shape) -> Quadrature:
+    """
+    :returns: a :class:`~modepy.Quadrature` that exactly integrates the functions
+        in *space*.
+    """
+    raise NotImplementedError((type(space).__name__, type(shape).__name))
+
+
+@quadrature_for_space.register(PN)
+def _(space: PN, shape: Simplex):
+    if not isinstance(shape, Simplex):
+        raise NotImplementedError((type(space).__name__, type(shape).__name))
```

Comment:
If only Python had multiple dispatch like Julia :cry:

Reply:
Agreed!

In fact, I briefly played with https://pypi.org/project/multipledispatch/ for this, but there were a number of (fixable!) shortcomings there that made me not like that particular realization. (around the `Dispatcher` and how they handle namespacing) Basically, if there were a multiple-dispatch package that works like `singledispatch`, I'd use it in a heartbeat.

At any rate, I figured we can easily incorporate that once a suitable solution shows up, starting from this interface.